bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexServerHandler.java

```diff
@@ -0,0 +1,306 @@
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlexServerHandler} is responsible for creating the
+ * Bridge Thing for a PLEX Server.
+ *
+ * @author Brian Homeyer - Initial contribution
+ */
+@NonNullByDefault
+public class PlexServerHandler extends BaseBridgeHandler implements PlexUpdateListener {
+
+    private final Logger logger = LoggerFactory.getLogger(PlexServerHandler.class);
+    private final PlexStateDescriptionOptionProvider stateDescriptionProvider;
+
+    // Maintain mapping of handler and players
+    private final Map<String, PlexPlayerHandler> playerHandlers = new ConcurrentHashMap<>();
+
+    private PlexServerConfiguration config = new PlexServerConfiguration();
+    // private @Nullable PlexServerHandler bridge;
+    private PlexApiConnector plexAPIConnector;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private volatile boolean isRunning = false;
+
+    public PlexServerHandler(Bridge bridge, PlexStateDescriptionOptionProvider stateDescriptionProvider) {
+        super(bridge);
+        plexAPIConnector = new PlexApiConnector(scheduler);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    public PlexApiConnector getPlexAPIConnector() {
+        return plexAPIConnector;
+    }
+
+    /**
+     * Initialize the Bridge set the config paramaters for the PLEX Server and
+     * start the refresh Job.
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(PlexServerConfiguration.class);
+        if (!EMPTY.equals(config.host)) { // Check if a hostname is set
+            plexAPIConnector.setParameters(config);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Host must be specified, check configuration");
+            return;
+        }
+        if (!plexAPIConnector.hasToken()) {
+            // No token is set by config, let's see if we can fetch one from username/password
+            logger.debug("Token is not set, trying to fetch one");
+            if ((EMPTY.equals(config.getUsername()) || EMPTY.equals(config.getPassword()))) {
+                logger.debug("Username, password and Token is not set, unable to connect to PLEX without. ");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Username, password and Token is not set, unable to connect to PLEX without. ");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Please check all.